openbas-api/src/main/java/io/openbas/utils/InjectMapper.java

```diff
@@ -97,23 +96,93 @@ private PayloadSimple toPayloadSimple(Optional<Payload> payload) {
+  }
+
   public InjectStatusOutput toInjectStatusOutput(Optional<InjectStatus> injectStatus) {
     return injectStatus
         .map(
             status ->
-                InjectStatusOutput.builder()
-                    .id(status.getId())
-                    .name(
-                        Optional.ofNullable(status.getName())
-                            .map(ExecutionStatus::name)
-                            .orElse(null))
-                    .traces(status.getTraces())
-                    .trackingSentDate(status.getTrackingSentDate())
-                    .trackingEndDate(status.getTrackingEndDate())
-                    .build())
+                this.<InjectStatusOutput>buildInjectStatusOutput(
+                    InjectStatusOutput.builder().build(), status, status.getTraces()))
         .orElseGet(() -> InjectStatusOutput.builder().build());
   }
 
+  public InjectTestStatusOutput toInjectTestStatusOutput(InjectTestStatus injectTestStatus) {
+    InjectTestStatusOutput output = InjectTestStatusOutput.builder().build();
+    buildInjectStatusOutput(output, injectTestStatus, injectTestStatus.getTraces());
+
+    output.setInjectId(injectTestStatus.getInject().getId());
+    output.setInjectType(
+        injectTestStatus
+            .getInject()
+            .getInjectorContract()
+            .map(InjectorContract::getInjector)
+            .map(Injector::getType)
+            .orElse(null));
+    output.setInjectTitle(injectTestStatus.getInject().getTitle());
+
+    return output;
+  }
+
+  public List<ExecutionTracesOutput> toExecutionTracesOutput(List<ExecutionTraces> traces) {
+    return traces.stream()
+        .map(
+            trace ->
+                ExecutionTracesOutput.builder()
+                    .status(trace.getStatus())
+                    .time(trace.getTime())
+                    .message(trace.getMessage())
+                    .action(trace.getAction())
+                    .build())
+        .toList();
+  }
+
+  private List<AgentStatusOutput> groupTracesByAgent(List<ExecutionTraces> traces) {
+    return traces.stream()
+        .collect(Collectors.groupingBy(ExecutionTraces::getAgent))
+        .entrySet()
+        .stream()
+        .map(
+            entry -> {
+              ExecutionTraces finalTrace =
+                  entry.getValue().stream()
+                      .filter(t -> t.getAction() == ExecutionTraceAction.COMPLETE)
+                      .findFirst()
+                      .orElse(null);
+              Agent agent = entry.getKey();
+              return AgentStatusOutput.builder()
+                  .assetId(agent.getAsset().getId())
+                  .agentId(agent.getId())
+                  .agentExecutorName(agent.getExecutor().getName())
+                  .agentExecutorType(agent.getExecutor().getType())
+                  .agentName(agent.getExecutedByUser())
+                  .statusName(finalTrace != null ? String.valueOf(finalTrace.getStatus()) : null)
```

Comment:
If finalTrace.getStatus() is an enum, replace String.valueOf(finalTrace.getStatus()) by finalTrace.getStatus().name()